sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -172,6 +172,18 @@ object TypeCoercion {
     case _ => None
   }
 
+  /**
+   * The method finds a common type for data types that differ only in nullable, containsNull
+   * and valueContainsNull flags. If the input types are too different, None is returned.
+   */
+  def findCommonTypeDifferentOnlyInNullFlags(t1: DataType, t2: DataType): Option[DataType] = {
```

Comment:
why define the method here as it's only used in `ComplexTypeMergingExpression`?

Reply:
Actually I'd like to use it at #21704.